### PR TITLE
feat(editor): add template switcher 

### PR DESCRIPTION
**Pull request by PatrickDohn**
#### Summary

- Work Type: Feature

#### What I did

- created template switcher to switch between templates
- added modal for viewing template options
- added React Modal package

#### Preview

https://5fc8502a23719100092d3f07--res-gen.netlify.app/

#### Checklist

- [x] Branch is in format `TYPE/scope/description`
- [x] PR title is in format `TYPE(scope): description`
- [x] Commits are in semantic format
- [x] Has Summary
- [x] Has Labels

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
+    "bootstrap": "^4.5.3",
```
Comment (davidholyko) [inline package.json:10]: remove this package :)

File: src/components/pages/EditorPage.js [hunk 2]
```diff
@@ -34,7 +38,53 @@ export const EditorPage = () => {
...
+      right: 'auto',
+      bottom: 'auto',
+      marginRight: '-50%',
+      transform: 'translate(-50%, -50%)',
+    },
+  };
+  var subtitle;
+
+  const [modalIsOpen, setIsOpen] = React.useState(false);
+  function openModal() {
```
Comment (davidholyko) [inline src/components/pages/EditorPage.js:54]: change to arrow

File: src/components/pages/EditorPage.js [hunk 3]
```diff
@@ -34,7 +38,53 @@ export const EditorPage = () => {
...
+    setIsOpen(true);
+  }
+
+  function afterOpenModal() {
```
Comment (davidholyko) [inline src/components/pages/EditorPage.js:58]: change to arrow

File: src/components/pages/EditorPage.js [hunk 4]
```diff
@@ -34,7 +38,53 @@ export const EditorPage = () => {
...
+    // references are now sync'd and can be accessed.
+    subtitle.style.color = 'black';
+  }
+
+  function closeModal() {
```
Comment (davidholyko) [inline src/components/pages/EditorPage.js:63]: change to arrow

File: src/components/pages/EditorPage.js [hunk 5]
```diff
@@ -34,7 +38,53 @@ export const EditorPage = () => {
...
+    content: {
+      top: '50%',
+      left: '75%',
+      right: 'auto',
+      bottom: 'auto',
+      marginRight: '-50%',
+      transform: 'translate(-50%, -50%)',
+    },
+  };
+  var subtitle;
```
Comment (davidholyko) [inline src/components/pages/EditorPage.js:51]: use react useRef

File: src/components/pages/EditorPage.js [hunk 6]
```diff
@@ -34,7 +38,53 @@ export const EditorPage = () => {
     },
   ];
 
+  const customStyles = {
```
Comment (davidholyko) [inline src/components/pages/EditorPage.js:41]: use className with scss

File: src/components/pages/EditorPage.js [hunk 7]
```diff
@@ -66,7 +116,32 @@ export const EditorPage = () => {
...
+        <button onClick={openModal}>Open Modal</button>
+        {renderTemplate(resumeIndex)}
+      </div>
+
+      <div>
+        <Modal
+          isOpen={modalIsOpen}
+          onAfterOpen={afterOpenModal}
+          onRequestClose={closeModal}
+          style={customStyles}
```
Comment (davidholyko) [inline src/components/pages/EditorPage.js:128]: use className with scss

File: src/components/pages/EditorPage.js [hunk 8]
```diff
@@ -66,7 +116,32 @@ export const EditorPage = () => {
...
+          contentLabel="Example Modal"
```
Comment (davidholyko) [inline src/components/pages/EditorPage.js:129]: rename content to something meaningful, Template Switcher Modal?

File: src/components/pages/EditorPage.js [hunk 9]
```diff
@@ -66,7 +116,32 @@ export const EditorPage = () => {
...
+        >
+          <h2 ref={(_subtitle) => (subtitle = _subtitle)}>Select a template</h2>
```
Comment (davidholyko) [inline src/components/pages/EditorPage.js:131]: use React ref

File: src/components/pages/EditorPage.js [hunk 10]
```diff
@@ -66,7 +116,32 @@ export const EditorPage = () => {
...
+          <button onClick={closeModal}>Exit</button>
+          <div>
+            {switchResume.map((resume, resumeIndex) => (
+              <button
+                className="imgButton"
```
Comment (davidholyko) [inline src/components/pages/EditorPage.js:136]: kebab case

File: src/components/pages/EditorPage.js [hunk 11]
```diff
@@ -66,7 +116,32 @@ export const EditorPage = () => {
...
+                key={resumeIndex}
+                onClick={() => setResumeIndex(resumeIndex)}
+              >
+                <img className="coverImg" src={resume.image} />
```
Comment (davidholyko) [inline src/components/pages/EditorPage.js:140]: kebab case

File: src/components/pages/EditorPage.js (hunk 7 above)
Comment (PatrickDohn) [reply src/components/pages/EditorPage.js:133]: Modal styles won't work without custom styles object

File: src/components/customizers/EditorCustomizer.js [hunk 12]
```diff
@@ -21,11 +20,11 @@ export const EditorCustomizer = ({ setData, defaultData, type }) => {
     );
     setEditor(codeMirrorEditor);
     setData(defaultData);
-  }, [type, editorRef]);
+  }, [type, editorRef, defaultData]);
```
Comment (davidholyko) [inline src/components/customizers/EditorCustomizer.js:23]: fixes bug where resetting defaultData triggered by switch does not re render Editor JSON

**Review by davidholyko — APPROVED** [review]
LGTM

**Pull request merged**